include/exadg/grid/grid.h

```diff
@@ -68,6 +99,23 @@ class Grid
    * dealii::Mapping.
    */
   std::shared_ptr<dealii::Mapping<dim>> mapping;
+
+  /**
+   * a vector of coarse triangulations required for global coarsening multigrid
+   */
+  std::vector<std::shared_ptr<dealii::Triangulation<dim> const>> coarse_triangulations;
+
+  /**
+   * a vector of dealii::GridTools::PeriodicFacePair's for the coarse triangulations required for
+   * global coarsening multigrid.
+   */
+  std::vector<PeriodicFaces> coarse_periodic_faces;
+
+  /**
+   * a vector of dealii::Mapping's for the coarse triangulations required for global coarsening
+   * multigrid.
+   */
+  std::vector<std::shared_ptr<dealii::Mapping<dim>>> coarse_mapping;
```

Comment:
Do you plan to switch these three variables to
```c++
  std::vector<Grid<dim>> coarse_grids;
```
in the future?

Reply:
happening right now

Reply:
Things turned out much more complicated than I thought yesterday. "Hiding" mapping inside `Grid` inside `GridManager` is not compatible with the Moving Mesh (ALE) functionality in ExaDG, which would need some major refactoring that should not get mixed with the intention of the present PR, namely allowing periodic faces for global coarsening multigrid. I think we need to do things step-by-step and for now just insert a vector of PeriodicFaces for all the coarse triangulations into the class `Grid`. Five hours of making changes to the code indicate once more to work as much as possible on plain deal.II data types since this allows most stable interfaces, even though they might be growing over time (like the one for our multigrid implementation).